File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/notice/CashshopNoticeListItemDTO.java
package dev.spiralmoon.maplestory.api.dto.notice;

import com.google.gson.annotations.SerializedName;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.time.LocalDateTime;

/**
 * 캐시샵 공지
 */
@AllArgsConstructor
@Data
@ToString
public class CashshopNoticeListItemDTO {

    /**
     * 공지 제목
     */
    @SerializedName("id")
    private String title;

    /**
     * 공지 링크
     */
    @SerializedName("url")
    private String url;

    /**
     * 공지 식별자
     */
    @SerializedName("notice_id")
    private int noticeId;

    /**
     * 공지 등록일
     */
    @SerializedName("date")
    private String date;

    /**
     * 판매 시작일
     */
    @SerializedName("date_sale_start")
    private String dateSaleStart;

    /**
     * 판매 종료일
     */
    @SerializedName("date_sale_end")
    private String dateSaleEnd;

    /**
     * 상시 판매 여부 (true - 상시)
     */
    @SerializedName("ongoing_flag")
    private String ongoingFlag;

    /**
     * 공지 등록일
     */
    public LocalDateTime getDate() {
        return Utils.toLocalDateTime(this.date);
    }

    /**
     * 판매 시작일
     */
    public LocalDateTime getDateSaleStart() {
        return this.dateSaleStart != null ? Utils.toLocalDateTime(this.dateSaleStart) : null;
    }

    /**
     * 판매 종료일
     */
    public LocalDateTime getDateSaleEnd() {
        return this.dateSaleEnd != null ? Utils.toLocalDateTime(this.dateSaleEnd) : null;
    }

    /**
     * 상시 판매 여부 (true - 상시)
     */
    public boolean isOnGoing() {
        return "true".equals(this.ongoingFlag);
    }
}

File: js/src/maplestory/api/dto/notice/cashshopNoticeListItemDto.ts
import { CashshopNoticeListItemDtoBody } from '../../response/notice/cashshopNoticeListBody';

/**
 * 캐시샵 공지
 */
class CashshopNoticeListItemDto {

  /**
   * 공지 제목
   */
  title: string;

  /**
   * 공지 링크
   */
  url: string;

  /**
   * 공지 식별자
   */
  noticeId: number;

  /**
   * 공지 등록일
   */
  date: Date;

  /**
   * 판매 시작일
   */
  dateSaleStart: Date | null;

  /**
   * 판매 종료일
   */
  dateSaleEnd: Date | null;

  /**
   * 상시 판매 여부 (true - 상시)
   */
  ongoingFlag: string;

  constructor(obj: CashshopNoticeListItemDtoBody) {

    const { title, url, notice_id, date, date_sale_start, date_sale_end, ongoing_flag } = obj;

    this.title = title;
    this.url = url;
    this.noticeId = notice_id;
    this.date = new Date(date);
    this.dateSaleStart = date_sale_start ? new Date(date_sale_start) : null;
    this.dateSaleEnd = date_sale_end ? new Date(date_sale_end) : null;
    this.ongoingFlag = ongoing_flag;
  }


  /**
   * 상시 판매 여부 (true - 상시)
   */
  get isOnGoing() {
    return this.ongoingFlag === 'true';
  }
}

export { CashshopNoticeListItemDto };
